lib/ansible/template/vars.py

```diff
@@ -69,6 +71,14 @@ def __contains__(self, k):
             return True
         return False
 
+    def __iter__(self):
+        keys = set()
+        keys.update(self._templar._available_variables, self._locals, self._globals, *self._extras)
+        return iter(keys)
+
+    def __len__(self):
+        return sum(1 for k in self.__iter__())
```

Comment:
This will be a little slow (how apparent that is depends on how many variables a site has defined).  Even though it's not-DRY, Probably better to copy the ```__iter_```` code that creates the keys  here and then ```return len(keys)```